src/dial.js

```diff
@@ -404,7 +408,8 @@ class Dialer {
         return callback(err)
       }
 
-      selectSafe(msDialer, this.protocol, (err, conn) => {
+      selectSafe(msDialer, this.protocol, (err, _conn) => {
+        const conn = observeConnection(null, this.protocol, _conn, this.switch.observer)
```

Comment:
prolly need to abort on error here

Reply:
Actually, it seems like the error just gets propagated through the `callback` in the following line.

Reply:
im mean that the `observeConnection` call might break if we pass in an undefined `_conn` in the event of an `err`

Reply:
the ol `if (err) return cb(err)` abort statement

Reply:
you are right, that part should be more standard anyways.